### DISCUSSION TITLE
Add Attachment Mapping File support

### DISCUSSION THREAD

**Discussion by grantfitzsimmons** [specify/specify7#5941] [Feature Requests]
**Is your feature request related to a problem? Please describe.**
>Yes, the catalog number field is a text field in several of the UMich databases. This enables support for the non-number characters they store in the field, like the UMMZI- suffix in this case. 
>
> In S6, we have been using an attachment mapping file which describes the catalog numbers and attachment filenames explicitly. I'm attaching an example here for reference. If a similar system could be added to S7, that would be great.

In this example, this mapping file is necessary to work around an issue with the catalog number field being formatted.

See this video to see how it worked: https://discourse.specifysoftware.org/t/attachment-mapping-file-video/101?u=grant

From @realVinayak:
> Yes, unless the catalog number field itself is formatted (has a formatter other than text), then multiple records would not be matched. Specify 6 "got around" the issue via the mapping file

Here's an example file sent to us by Garth at UMich:
**Example mapping file:**
> |Catalog Number|Attachment Name|
> |---|---|
> |UMMZI-90126|UMMZI-90126.jpg|
> |UMMZI-90159|UMMZI-90159.jpg|
> |UMMZI-90171|UMMZI-90171.jpg|
> |UMMZI-90175|UMMZI-90175_1.jpg|
> |UMMZI-90175|UMMZI-90175.jpg|
> |UMMZI-90176|UMMZI-90176_1.jpg|
> |UMMZI-90176|UMMZI-90176.jpg|
> |UMMZI-90179|UMMZI-90179_1.jpg|
> |UMMZI-90179|UMMZI-90179.jpg|
> |UMMZI-90209|UMMZI-90209_1.jpg|
> |UMMZI-90209|UMMZI-90209.jpg|
> |UMMZI-90231|UMMZI-90231_1.jpg|
> |UMMZI-90231|UMMZI-90231.jpg|
> |UMMZI-90234|UMMZI-90234_1.jpg|
> |UMMZI-90234|UMMZI-90234.jpg|
> |UMMZI-90237|UMMZI-90237.jpg|
> |UMMZI-119907|UMMZI-119907_1.jpg|
> |UMMZI-119907|UMMZI-119907.jpg|
> |UMMZI-119908|UMMZI-119908_1.jpg|
> |UMMZI-119908|UMMZI-119908.jpg|

**Describe the solution you'd like**
It should be implemented as it was in Specify 6.

**Describe alternatives you've considered**
They tried to import using the batch attachment uploader but could not indicate multiple images associated with a single specimen due to the formatting for that field. 

**Reported By**
Garth at University of Michigan

**Comment by specifysoftware** [comment]
This issue has been mentioned on **Specify Community Forum**. There might be relevant details there:

https://discourse.specifysoftware.org/t/batch-attachment-uploader/1374/5

**Comment by grantfitzsimmons** [comment]
Queensland Herbarium has requested this capability as well

**Comment by grantfitzsimmons** [comment]
Mentioned by Michigan today

**Comment by totallynotleo** [comment]
I know that Paul Robins already requested this for the Queensland Herbarium but as we're looking into Specify 6 EOL preparation and moving to Specify 7 fully not being able to use an attachment mapping is one of the main blockers, so I figured it wouldn't hurt to repost it again. The alternative for uploading multiple attachments means we have to change our naming structure from BRI-AQ123456_1.jpeg to 000123456_1.jpeg which would break a lot of our back end image backup and automation processes, and cause issues with the >300000 images that are named like this already. Keeping the Herbarium identifier is very useful as these images are also the images shared on multiple platforms.

**Comment by grantfitzsimmons** [comment]
Added in https://github.com/specify/specify7/pull/8311 and will be in the next update!
